xilem_core/src/views/memoize.rs

```diff
@@ -64,17 +64,20 @@ where
         (element, memoize_state)
     }
 
-    fn rebuild(
+    fn rebuild<'a>(
         &self,
         prev: &Self,
         view_state: &mut Self::ViewState,
         cx: &mut Context,
-        element: <Self::Element as ViewElement>::Mut<'_>,
-    ) {
+        element: <Self::Element as ViewElement>::Mut<'a>,
+    ) -> <Self::Element as ViewElement>::Mut<'a> {
```

Comment:
I wonder if we can add a `xilem_core::Mut` type alias, to remove the `as ViewElement`.

Reply:
I've tested it with the following which does work:
```rust
pub type Mut<'a, E> = <E as ViewElement>::Mut<'a>;
```
it resolves to something like when using rust-analyzer and using it in the trait: 
```suggestion
    ) -> xilem_core::Mut<'a, Self::Element> {
```

Should I add that type alias?

Reply:
I think that would be good. I'd hope to remove the `xilem_core::` as well by importing it.

If it were in a seperate PR, I'd just merge that very quickly; I'm still not completely sure I want this change, although now is the time to make it...

Reply:
Yes that's possible with the code-action to remove that `xilem_core::`, and when it's in scope (`use xilem_core::Mut`) it won't appear in the code-action to impl that `View` trait.

I get that hesitation, but I do think it's worth it, since it's basically all autogenerated boilerplate (via rust-analyzer) and the lifetime is somewhat trivial, but it adds additional flexibility (also in the main xilem implementation) by being able to modify the element after descendent views have modified it, which I think is quite useful for modifier based implementations that rely on previous modifications (while not having `&mut E` as `ViewElement::Mut` type).